storage/innobase/xtrabackup/src/xtrabackup.cc

```diff
@@ -1532,6 +1531,15 @@ check_if_param_set(const char *param)
 	return param_set.find(param) != param_set.end();
 }
 
+/* distory the argument */
+static void hide_option(char *argument, char **opt) {
+	char *start = argument;
+	my_free(*opt);
+	*opt = my_strdup(PSI_NOT_INSTRUMENTED,argument, MYF(MY_FAE));
+	while (*argument) *argument++ = 'x'; /*Destroy argument */
+	if (*start) start[1] = 0;   /*cut lenght of argument */
```

Comment:
Should we truncate the string first? and then assign x?

Reply:
if somebody can access argument in some way. so better would be to destroy and truncate. so we sure it not in memory.

Reply:
ok please make it 'xxxx' instea of 'x xxxx'